test/Microsoft.OpenApi.Readers.Tests/V31Tests/OpenApiSchemaTests.cs

```diff
@@ -542,6 +542,43 @@ public void ParseSchemaWithUnrecognizedKeywordsWorks()
             Assert.Equal(2, schema.UnrecognizedKeywords.Count);
         }
 
+        [Fact]
+        public void ParseSchemaExampleWithPrimitivesWorks()
+        {
+            var expected1 = @"{
+  ""type"": ""string"",
+  ""example"": ""2024-01-02""
+}";
+
+            var expected2 = @"{
+  ""type"": ""string"",
+  ""example"": ""3.14""
+}";
+            var schema = new OpenApiSchema()
+            {
+                Type = JsonSchemaType.String,
+                Example = JsonValue.Create("2024-01-02")
```

Comment:
we should also have a unit test where the actual instance is a DateTime instance. As I'm not sure TryGetValue in that case will return a string. See https://github.com/microsoft/OpenAPI.NET/pull/2157/files#diff-0f3a7509d4954ac2bbc88c7eb76ef6dca6b526cef456fa0d7610013421ce7f96R77 for more context